vmdb/app/models/job_proxy_dispatcher.rb

```diff
@@ -142,20 +146,18 @@ def assign_proxy_to_job(proxy, job)
 
   def pending_jobs
     @zone = MiqServer.my_zone
-    Job.find(:all, :order => "id" ,:conditions => [
-        "state = ? and dispatch_status = ? and
-     (zone is null or zone = ?) and
-     (sync_key is NULL or
-       sync_key not in (
-         select sync_key from jobs where
-           dispatch_status = 'active' and
-           state != 'finished' and
-           (zone is null or zone = ?) and
-           sync_key is not NULL
-       )
-    )",
-        "waiting_to_start", "pending", @zone, @zone
-      ])
+    Job.order(:id)
+      .where(:state           => "waiting_to_start")
+      .where(:dispatch_status => "pending")
+      .where("zone is null or zone = ?", @zone)
+      .where("sync_key is NULL or
+        sync_key not in (
+           select sync_key from jobs where
+             dispatch_status = 'active' and
+             state != 'finished' and
+             (zone is null or zone = ?) and
+             sync_key is not NULL)", @zone)
+      .to_a
```

Comment:
I'm going to remove these .to_a calls, let's try to use relations association proxies as much as possible.